plasmapy/formulary/shocks.py

```diff
@@ -0,0 +1,116 @@
+""" This module gathers functions relating to shocks and the properties thereof.
+"""
+__all__ = ["entropy_jump_polytropic"]
+
+
+import warnings
+from astropy import units as u
+from plasmapy.utils.exceptions import PhysicsWarning
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log, nan, isclose
+
+
+@validate_quantities(
+    rho_1={"can_be_negative": False, "pass_equivalent_units": True},
+    rho_2={"can_be_negative": False, "pass_equivalent_units": True},
+    p_1={"can_be_negative": True, "pass_equivalent_units": True},
+    p_2={"can_be_negative": True, "pass_equivalent_units": True},
+    c_v={"can_be_negative": False},
+)
+def entropy_jump_polytropic(
```

Comment:
I have problem with this name, it's not explicit enough when imported from the `fomulary` module.  That is, this makes sense...

```python
from plasmapy.formulary import shocks

val = shocks.entropy_jump_polytropic(...)
```
but this does not make sense
```python
from plasmapy.formulary import entropy_jump_polytropic

val = entropy_jump_polytropic(...)
```

In the later case, we have no indication that the function is purposefully for shocks.  I see three possible solutions...

1. Rename the function to be more explicit. (e.g `entropy_jumpy_polytropic_gas_shock`)
1. Keep the function name  as is, but give it an more explicit name when imported into `plasmapy/formulary/__init__.py`
1. Only expose `shocks` to the formulary namespace. (i.e. you would have to import like `from plasmapy.formulary.shocks import entropy_jumpy_polytropic`)

My opinion, option 3 is the best choice.  @StanczakDominik @namurphy do you have any thoughts about this?

Reply:
@rocco8773 This is resolved now, right?

Reply:
Actually, maybe it's worth going one further: since this function only applies to shocks described by a certain model, maybe the namespace should be more like: `formulary.shocks.polytropic.entropy_jump`?

Reply:
If we're going that route, I'd go with `shocks.gas.entropy_jump_polytropic` since the expression is derived from a purely thermodynamic viewpoint.

This has always been one of areas of confusion on this topic, at least for me.  We have sets of equations that are derived from a purely thermodynamic viewpoint, which means the fluid used in the derivation is gas-like.  And then we have equations that take into consideration charges, currents, and fields, these are more representative of a plasma.  I think the field has always done a bad job of distinguishing the difference between these expressions, and that's why I'm so stuck up on the name here.  I'd like to take this opportunity to use `plasmapy` to help clarify this point and reduce the confusion in the community.

With that said, I still think the best name so far is `entropy_jumpy_polytropic_gas`.  @pheuer If you have arguments for or against this, then I'd love to hear them!?